library/model/book.py

```diff
@@ -168,3 +136,68 @@ def serialize(self, book: Book, format: str):
             return et.tostring(book_info, encoding="Unicode")
         else:
             raise ValueError(format)
+
+
+class ElectronicBook(Book):
+    pages: int
+
+    def __init__(self, title, authors, publisher, pub_date, genres, pages, isbn):
+        super(ElectronicBook, self).__init__(title, authors, publisher, pub_date, genres, isbn)
+        self.pages = pages
+
+    def can_borrow(self) -> bool:
+        return True
+
+    def get_approximate_duration(self) -> int:
+        return self.pages * 5 * 60
+
+    def get_weekly_fee(self) -> int:
+        return 2
+
+
+class AudioBook(Book):
+    duration: int
+
+    def __init__(self, title, authors, publisher, pub_date, genres, isbn, duration=0):
+        super(AudioBook, self).__init__(title, authors, publisher, pub_date, genres, isbn)
+        self.duration = duration
+
+    def can_borrow(self) -> bool:
+        return True
+
+    def get_approximate_duration(self) -> int:
+        return self.duration
+
+    def get_weekly_fee(self) -> int:
+        return 2
+
+
+class PaperBook(Book):
+
+    existing_items: int
+    borrowed_items: int
+    pages: int
+
+    def __init__(self, title, authors, publisher, pub_date, genres, pages, isbn, existing_items=1, borrowed_items=0):
+        super().__init__(title, authors, publisher, pub_date, genres, isbn)
```

Comment:
inconsisten use of super() argument [here, without arguments - in AudioBook, with arguments]